async_simple/uthread/test/UthreadTest.cpp

```diff
@@ -111,9 +111,9 @@ TEST_F(UthreadTest, testSwitch) {
         Promise<int> p;
         auto f = p.getFuture().via(&_executor);
         delayedTask(
-            [p = std::move(p)]() mutable {
+            [wrapper = MoveWrapper(std::move(p))]() mutable {
```

Comment:
I feel such change would make it harder to use. I am sure seastar has copy constructor/assignment for promise before. I am wondering if it is better to fix the copy constructor/assignment **now**. And delete it later when we introduce `noncopyable_function `.

Reply:
we can create another issue to remove MoveWrapper, maybe use` noncopyable_function` or the other solution

Reply:
FutureState.h is internal too. It doesn't matter if we write some not direct code. But now user need to write `MoveWrapper` in some cases, it is not straightforward. I prefer to fix the copy constructor/assignment in this PR and create an issue to introduce `nocopyable_function` and delete copy constructor/assignment.

Reply:
@foreverhy how do you think?

Reply:
Actually i tried to remove MoveWrapper before, replace it with move capture, but compile failed, maybe need to do more changes.
I wish we can use move capture instead of MoveWrapper.

Reply:
emm, I agree to fix the copy assignment operator firstly